gluonnlp/data/transforms.py

```diff
@@ -668,3 +668,131 @@ def __call__(self, sample):
             List of tokens
         """
         return [tok for tok in self._tokenizer.segment(sample).strip().split()]
+
+
+class SentencepieceTokenizer(object):
+    r"""Apply the Sentencepiece Tokenizer, which supports subword tokenization such as BPE.
+
+    Users of this class are required to `install sentencepiece
+    <https://github.com/google/sentencepiece>`_. For example, one can use:
+
+    .. code:: python
+
+        pip install -U sentencepiece
+
+    Parameters
+    ----------
+    path : str
+        Path to the pre-trained subword tokenization model.
+
+    Examples
+    --------
+    >>> tokenizer = SentencepieceTokenizer('./corpus.full.bpe')
+    >>> detokenizer = SentencepieceDetokenizer('./corpus.full.bpe')
+    >>> sentence = 'This is a very awesome, life-changing sentence.'
+    >>> tokenizer(sentence)
+    ['▁This',
```

Comment:
This looks different from the moses BPE, in which subwords are identified by symbols @@. For example, with moses BPE, '_This' will be 'This' and 'ch' will be 'ch@@'.

Reply:
Yes indeed, and I wasn't able to find an option to set the word boundary markers. Is there such option in moses bpe?

Reply:
On the other hand, we may choose to integrate https://github.com/rsennrich/subword-nmt to support both, which carries an MIT license.

Reply:
I will address this in a separate PR.